src/Psy/Command/SandboxCommand.php

```diff
@@ -0,0 +1,269 @@
+                ),
+                new InputArgument(
+                    'name',
+                    InputArgument::OPTIONAL,
+                    'Optional name for the sandbox'
+                ),
+            ))
+            ->setDescription('Manages sandboxed environments.')
+            ->setHelp(
+                <<<HELP
+sandbox add [name] creates a encapsulated sandbox with an optional name, if not provided will autogenerate a name.
+sandbox list Will show a list of the currently generated sandboxes
+sandbox which Will show the current sandbox
+sandbox switch name Switches to sandbox by name,
+sandbox delete [name] Removes a sandbox by name, or the current if name is not provided.
+sandbox exit Erases all sandboxes and sets the normal state.
+HELP
+            );
+
+        $this->restoreFolder = exec('pwd');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->input = $input;
+        $this->output = $output;
+
+        $action = $this->input->getArgument('action');
+        $name = $this->input->getArgument('name');
+        if (!in_array($action, $this->allowedActions)) {
+            throw new \InvalidArgumentException("Action $action is not allowed.");
+        }
+
+        $this->performAction($action, $name);
+    }
+
+    /**
+     * @param $action
+     * @param $name
+     */
+    protected function performAction($action, $name)
+    {
+        switch ($action) {
+            case self::ACTION_ADD:
+                $identifier = $this->createSandbox($name);
+                $this->switchSandbox($identifier);
+                break;
+            case self::ACTION_LIST:
+                $this->listSandboxes();
+                break;
+            case self::ACTION_WHICH:
+                $this->showCurrentSandbox();
+                break;
+            case self::ACTION_SWITCH:
+                $this->switchSandbox($name);
+                break;
+            case self::ACTION_DELETE:
+                $this->removeSandbox($name);
+                break;
+            case self::ACTION_EXIT:
+                $this->restoreState();
+                break;
+        }
+    }
+
+    /**
+     * @param $name
+     * @return string
+     */
+    protected function getSandboxPath($name)
+    {
+        return sys_get_temp_dir() . DIRECTORY_SEPARATOR . self::SANDBOX_FOLDER . DIRECTORY_SEPARATOR . $name;
+    }
+
+    /**
+     * @param  null        $name
+     * @return null|string
+     */
+    protected function createSandbox($name = null)
+    {
+        $name = !is_null($name) ? $name : uniqid();
+        $path = $this->getSandboxPath($name);
+        if (file_exists($path)) {
+            throw new \InvalidArgumentException("Sandbox with name : $name already exists.");
+        }
+
+        $this->output->writeln("<info>Trying to create sandbox named : $name</info>");
```

Comment:
we don't need this output, it's just saying "trying to do what you asked", after which it'll say either "did what you asked" or "didn't do what you asked"… so the trying bit is redundant :)